### PR TITLE
submitted an initial solution to week 8

### PR DESCRIPTION
**Pull request by Andrey8860**

### CODE REVIEW

**Comment by Andrey8860** [comment]
I have added a new commit which includes some fixes, including those which were highlighted for week 7. I also added a video of tests running on Selenium Grid. Video also includes a custom config file for one node which is not actually used - this was supposed to be used for Sauce Labs, however I ran into configuration issues which due to a lack of time I decided to skip, since I got the general idea of how Sauce Labs should be used. However, I included this into the video to show the work which was actually done and finished.

File: Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+
+import com.codeborne.selenide.WebDriverRunner;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class BaseTest {
+	
+	private static final String HUB_URL = "http://192.168.0.36:4444";
```
Comment (mamax) [inline Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java:22]: need to move this to some property file

File: Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		driver = new ChromeDriver();
+		driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(3));
+		driver.manage().window().maximize();
+	}
+
+	private void setUpRemote(String browser, String platform) throws MalformedURLException {
+		DesiredCapabilities capabilities = new DesiredCapabilities();
+		capabilities.setBrowserName(browser);
+		
+		switch(platform.toLowerCase()) {
```
Comment (mamax) [inline Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java:56]: `Switch - case` with one case is not a good approach. Please add more cases for different platforms, if you want to use `switch-case`

File: Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		if(driver != null) {
+			driver.quit();
+		}
+	}
+
+	private void setUpLocal() {
+		WebDriverManager.chromedriver().setup();
+		driver = new ChromeDriver();
+		driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(3));
+		driver.manage().window().maximize();
```
Comment (mamax) [inline Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java:49]: `implicitlyWait` & `window().maximize()` are duplicated in both `setUpLocal()` and `setUpRemote()` methods. Please make some separate methods or think about how to eliminate that duplication.

File: Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java (hunk 1 above)
Comment (Andrey8860) [reply Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java]: Done. Properties in this way were first introduced in module 13, so overall they are not used much for storing final variables, yet.

File: Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java (hunk 2 above)
Comment (Andrey8860) [reply Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java:56]: Done

File: Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java (hunk 3 above)
Comment (Andrey8860) [reply Week 8/solution/src/test/java/com/epam/week8/tests/BaseTest.java:49]: Done. Moved to another method. After this module implicit wait is not used at all so this should not happen again

**Pull request merged**